lib/utils.js

```diff
@@ -219,16 +219,58 @@ utils.findRefs = (json, absolutePath, relativePath, initialFormat, asyncapiYAMLo
 utils.getLocationOf = (jsonPointer, asyncapiYAMLorJSON, initialFormat) => {
   const ast = getAST(asyncapiYAMLorJSON, initialFormat);
   if (!ast) return { jsonPointer };
+
   return findLocationOf(jsonPointerToArray(jsonPointer), ast, initialFormat);
 };
 
 utils.improveAjvErrors = (errors, asyncapiYAMLorJSON, initialFormat) => {
   const ast = getAST(asyncapiYAMLorJSON, initialFormat);
   return errors.map(error => {
     const defaultLocation = { jsonPointer: error.dataPath || '/' };
+
     return {
       title: `${error.dataPath || '/'} ${error.message}`,
       location: ast ? findLocationOf(jsonPointerToArray(error.dataPath), ast, initialFormat) : defaultLocation,
     };
   });
-};
\ No newline at end of file
+};
+
+/**
+ * It parses the string and returns an array with all values that are between curly braces, including braces
+ */
+utils.parseUrlVariables = str => {
+  if (typeof str !== 'string') return;
+  const regex = new RE2(/{(.+?)}/g);
+
+  return regex.match(str); 
+};
+
+/**
+ * Returns an array of not existing properties in provided object with names specified in provided array
+ */
+utils.getMissingProps = (arr, obj) => {
+  arr = arr.map(val => val.replace(/[{}]/g, ''));
+
+  if (!obj) return arr;
+  
+  return arr.filter(val => {
+    return !obj.hasOwnProperty(val);
+  });
+};
+
+/**
+ * Returns array of errors for channel params and server variables validation
```

Comment:
This function is no longer **only** for channel params and server variables.

Reply:
what ya mean? at the moment it is used only for channels params and server variables. If we see it is useful in other validations then we should change it, or I'm missing the point?

Reply:
My point is this is the description of the function. It says what it does. IMO it doesn't have to say where it's used, given this can change in the future and it's easy to forget changing this description. I'd just make the description say "what this function does". Also, the `utils.js` file is for generic functions. If it's not generic, it should probably be in the `lib/customValidators.js` file. What do you think?

Reply:
ah, I know what you mean now, you're right, description should only say what function is doing of course. Will change, thanks. As if it comes to location, `utils.js` is fine as this function is already reused in two other functions and will be easier to notice in `utils.js`

Reply:
done